File: Competitive Programming 3/Section 1.3.3/10919 - Prerequisites?/C++/main.cpp
#include <bits/stdc++.h>

using namespace std;

bool taken[10000];

int main(){
    ios_base::sync_with_stdio(0);cin.tie(0);
    int k,m;
    while(cin>>k,k){
        cin>>m;
        memset(taken,false,sizeof(taken));
        for(int i = 0 ; i < k ; i ++){
            int number;
            cin >> number;
            taken[number]=true;
        }
        bool isYes=true;
        for(int i = 0 ; i < m; i ++){
            int c,r;
            cin >> c >> r ;
            for(int j = 0 ; j < c ; j ++){
                int number;
                cin >> number;
                if(taken[number]){
                    r--;
                }
            }
            if(r>0){
                isYes=false;
            }
        }
        cout << (isYes?"yes":"no") << "\n";
    }
    return 0;
}

File: Competitive Programming 3/Section 1.3.3/10919 - Prerequisites?/Java/Main.java
import java.util.Scanner;

public class Main{
    static Scanner s = new Scanner(System.in);
    static boolean[] taken;
    public static void main(String[]args){
        while(true){
            int k = s.nextInt();
            if(k==0)
                break;
            int m = s.nextInt();
            taken=new boolean[10000];
            for(int i = 0 ; i < k ; i ++){
                taken[s.nextInt()]=true;
            }   
            boolean isYes = true;         
            for(int i = 0 ; i < m ; i ++){
                int c=s.nextInt(),r=s.nextInt();
                for(int j = 0 ; j < c ; j ++){
                    if(taken[s.nextInt()]){
                        r--;
                    }
                }
                if(r>0)
                    isYes=false;
            }
            System.out.printf("%s\n",isYes?"yes":"no");
        }
    }
}